2-building-artifacts/2-docker.md

```diff
@@ -1,6 +1,83 @@
 # Running Go programs in Docker containers
 
-This should be done!
+Before looking at building Docker images, let's use it to compile go code
+
+## Compile using golang image
+Using golang image to build binaries can be useful for :
+- Continuous integration
+- Cross building binary
+- Local development
+
+### Continuous integration
+When using a CI server like Jenkins, it can become cumbersome to have to install all the go versions required by your different projects. And even more if you use your CI server to build projects in other languages, each with its own version (java8, java7, node, ruby, python, ...). For this use case, it can be easier to build the projects in docker containers. An example script could be :
+```
+ docker run --rm -v $PWD:/go/src/github.com/repo/project -w /go/src/github.com/repo/project golang:1.8.1 go build -o dist/binary
+```
+This will run the `go build` command with go version 1.8.1, sharing your current directory with the container. Obviously, you can run `go fmt/vet/test/...` commands the same way
+
+### Cross building binary
```

Comment:
hmmm... sorry don't understand what you mean here. The use case I have in mind in this section is only to use `docker run` to generate a binary, nothing more. Building an image  by adding the binary in the Dockerfile is shown as an example in the sections below.